include/boost/geometry/algorithms/detail/distance/default_strategies.hpp

```diff
@@ -110,26 +111,45 @@ struct default_strategy<Box1, Box2, box_tag, box_tag, false>
           >
 {};
 
+template <typename PolygonalOrLinear, typename Box>
+struct default_strategy_polygonal_or_linear
+    : strategy::distance::services::default_strategy
+        <
+            segment_tag, box_tag,
+            typename point_type<PolygonalOrLinear>::type,
+            typename point_type<Box>::type
+        >
+{};
+
 template <typename Linear, typename Box>
 struct default_strategy<Linear, Box, segment_tag, box_tag, false>
-    : strategy::distance::services::default_strategy
-          <
-              segment_tag, box_tag,
-              typename point_type<Linear>::type,
-              typename point_type<Box>::type
-          >
+    : default_strategy_polygonal_or_linear<Linear, Box>
```

Comment:
Doesn't it cause some cycle of base classes?

`default_strategy<segment, box>` is derived from `default_strategy_polygonal_or_linear` which is derived from `default_strategy<segment, box>` etc.

My gut feeling tells me that this particular specialization should define a strategy directly with `typedef`. Or am I missing something?

Reply:
What I'm missing is that this `default_strategy` is in the `namespace detail::distance`.